### ISSUE TITLE
Installation (Macbook Pro M3)

### ISSUE THREAD

**Issue by light-merlin-dark** [filipstrand/mflux#44]
First and foremost, thank you for such exquisite tech!
🤍🙏🏼🌹🪽✨ 

I'm using a Macbook Pro M3 and here's my experience:
1. Considered installing native but went for a python vm via `python3 -m venv ~/.ai`
2. Ran `pip install -U mflux`
3. Encountered the following error:
```
Collecting mflux
  Downloading mflux-0.2.0-py3-none-any.whl.metadata (19 kB)
INFO: pip is looking at multiple versions of mflux to determine which version is compatible with other requirements. This could take a while.
  Downloading mflux-0.1.9-py3-none-any.whl.metadata (16 kB)
ERROR: Cannot install mflux==0.1.9 and mflux==0.2.0 because these package versions have conflicting dependencies.

The conflict is caused by:
    mflux 0.2.0 depends on mlx>=0.16.0
    mflux 0.1.9 depends on mlx>=0.16.0

To fix this you could try to:
1. loosen the range of package versions you've specified
3. remove package versions to allow pip to attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflict
```

Considerations:
1. I assumed maybe it needed access to my native python libraries for the mlx dependency (not certain).
2. Maybe there was a recent build in your codebase of where this was derived

_*Update*:_

I just discovered that natively my M3 is using a version of python via rosetta, which is x86_64 cpu architecture and possibly where the error is occurring.

Either way, thanks for doing this - it's really cool and I'm looking forward to tinkering with it!

Salud!

🎩🌹🌙🪽✨

**Comment by filipstrand** [comment]
Hi @light-merlin-dark, I have now updated the readme with an additional instruction of setting up a new virtual environment before installing the project. If you follow these steps does it work for you?

**Comment by light-merlin-dark** [comment]
Hey @filipstrand good day!  

Aye, that was similar to the process I ran yesterday.  Only if I'm understanding correctly, it seems my native python env and the venv both are running via rosetta standard on my macbook m3, which may be significant in the context of this error and depending on which code runs internally to check e.g.:

`python3 -c "import platform; print(platform.machine())"` -> `x86_64`

`arm64`

Here's the output I get following the new `README`:

<img width="862" alt="Screenshot 2024-09-09 at 4 43 34 PM" src="https://github.com/user-attachments/assets/4500f1e7-0837-4dcf-9707-e3a143abf6d8">

I'm here to help test and try any new approach to resolve it either way.  Salud! 

🎩🌹☀️🪽✨

**Comment by awni** [comment]
You might find this article on [troubleshooting MLX installation](https://ml-explore.github.io/mlx/build/html/install.html#troubleshooting) useful. Depending on how you installed Python you may need to reinstall.

If you're using conda, something like:

```python
CONDA_SUBDIR=osx-arm64 conda create -n mflux
conda activate mflux
pip install mflux
```

should work in theory.

Should

**Comment by light-merlin-dark** [comment]
Hey, @awni.  Good day!  Aye, thank you!  That worked.  

For all clarity around this, it didn't seem my Macbook Pro M3 could run this out of the box without a way to get native python to recognize `arm` support.

Here's what I did to install with success:

1. I installed miniforge via Github [link here](https://github.com/conda-forge/miniforge)
2. `sh Miniforge3-MacOSX-arm64.sh`
3. `conda install conda-forge::mlx`
4. `CONDA_SUBDIR=osx-arm64 conda create -n mflux`
6. `conda activate mflux`
7. `conda install pip`
8. `pip install mflux`

Thanks again for everything.  I wish you all a wonderful day or evening!

🤍🙏🏼🌹🪽✨
